### PR TITLE
Docs: Add docs for nerdctl support

### PR DESCRIPTION
**Pull request by yankay**
After the #3429, the nerdctl is supported.

So add it to the docs.

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
/lgtm

**Comment by nunix** [comment]
hi everyone,

first of all, thanks a lot for this feature and the documentation of it 😄 

I did a quick test on the Finch WSL2 distro, and here's the output:
image

In the docs, I don't think there's the mention of the environment variable `KIND_EXPERIMENTAL_PROVIDER`, however as you can see, depending on when the docs will come out, we might need to mention it.

If you think/know this won't be needed as there's possibly an internal check, then please disregard my comment.

Once again, thanks a lot for the great work.

File: site/content/docs/contributing/development.md [hunk 1]
```diff
@@ -153,7 +153,7 @@ with kind at HEAD to have the latest fixes for running bleeding edge Kubernetes.
 We ensure that the tests continue to work in the kind repo before merging any code changes.
 
 We also have some limited usage experiments with [GitHub Actions], currently
-only for testing `podman` support in kind.
+only for testing `podman`, `nerdctl` support in kind.
```
[the remark this answers is not in the document]
Comment (yankay) [reply site/content/docs/contributing/development.md]: Thanks @stmcginnis

It has been changed.

**Comment by AkihiroSuda** [comment]
https://kind.sigs.k8s.io/docs/user/rootless/ should be updated too

**Comment by AkihiroSuda** [comment]
I also guess that `KIND_EXPERIMENTAL_PROVIDER` should be explained somewhere

**Comment by yankay** [comment]
> https://kind.sigs.k8s.io/docs/user/rootless/ should be updated too

Thanks it has been updated.

File: site/content/docs/user/rootless.md [hunk 2]
```diff
@@ -80,5 +81,12 @@ On some distributions, you might need to use systemd-run to start kind into its
 $ systemd-run --scope --user kind create cluster
 ```
 
+## Creating a kind cluster with Rootless Nerdctl
+
+To create a kind cluster with Nerdctl Podman, just run:
```
Comment (AkihiroSuda) [inline site/content/docs/user/rootless.md:86]:
```suggestion
To create a kind cluster with nerdctl, just run:
```

**Review by AkihiroSuda — APPROVED** [review]

File: site/content/docs/user/rootless.md [hunk 3]
```diff
@@ -6,11 +6,12 @@ menu:
     identifier: "rootless"
     weight: 3
 ---
-Starting with kind 0.11.0, [Rootless Docker](https://docs.docker.com/go/rootless/) and [Rootless Podman](https://github.com/containers/podman/blob/master/docs/tutorials/rootless_tutorial.md) can be used as the node provider of kind.
+Starting with kind 0.11.0, [Rootless Docker](https://docs.docker.com/go/rootless/), [Rootless Podman](https://github.com/containers/podman/blob/master/docs/tutorials/rootless_tutorial.md) and [Rootless Nerdctl](https://github.com/containerd/nerdctl/blob/main/docs/rootless.md) can be used as the node provider of kind.
 
 ## Provider requirements
 - Docker: 20.10 or later
 - Podman: 3.0 or later
+- Nerdctl: 1.7 or later
```
Comment (AkihiroSuda) [inline site/content/docs/user/rootless.md:14]: nit: maybe s/Nerdctl/nerdctl/g

**Comment by yankay** [comment]
> hi everyone,
> 
> first of all, thanks a lot for this feature and the documentation of it 😄
> 
> I did a quick test on the Finch WSL2 distro, and here's the output: image
> 
> In the docs, I don't think there's the mention of the environment variable `KIND_EXPERIMENTAL_PROVIDER`, however as you can see, depending on when the docs will come out, we might need to mention it.
> 
> If you think/know this won't be needed as there's possibly an internal check, then please disregard my comment.
> 
> Once again, thanks a lot for the great work.

Thanks @nunix 

Because I do have a finch WSL environment, I can only try to reproduce the issue in Ubuntu Linux.

```
root@kay201:~/oss/kind# docker version
-bash: /usr/local/bin/docker: No such file or directory
root@kay201:~/oss/kind# nerdctl version
Client:
 Version:       v1.7.2-102-gc18036e2.m
root@kay201:~/oss/kind# ./bin/kind --version
kind version 0.23.0-alpha.10+7c3e01fc1c25d8
root@kay201:~/oss/kind# ./bin/kind create cluster
Creating cluster "kind" ...
⠈⡱ Ensuring node image (kindest/node:v1.29.2) 🖼

```

It can work correctly without `KIND_EXPERIMENTAL_PROVIDER` , because the kind can auto-detect provider by https://github.com/kubernetes-sigs/kind/blob/main/pkg/cluster/provider.go#L114  .

Would you please help to provide the output with commands
```
nerdctl -v 
finch v
```
to help me invesgate the issue :-)

File: site/content/docs/user/rootless.md (hunk 3 above)
Comment (yankay) [reply site/content/docs/user/rootless.md]: Thanks, it has been updated.
And add some docs for `KIND_EXPERIMENTAL_PROVIDER` : https://github.com/kubernetes-sigs/kind/pull/3532/files#diff-de2961bf197d22d2e444cd3f30d82bbc479b53b49d99edcdec48045a2aae4c26

**Comment by nunix** [comment]
> nerdctl -v

@yankay certainly, here's the output (screenshot just for proof):
nerdctl version 1.7.3 (from Rancher Desktop install)
finch version v1.1.2

image

On the KinD side, I generated the binary from WSL2 Finch distro with the command `make install INSTALL_DIR=/usr/local/bin` (inside the git KinD directory).
The KinD version is:
kind v0.23.0-alpha.10+7c3e01fc1c25d8 go1.20.13 linux/amd64

Finally, I also built `kind.exe`, with the command `GOOS=windows go build` from WSL2.
The version is:
kind v0.23.0-alpha go1.21.7 windows/amd64

Same as before with `nerdctl`, I had to set `KIND_EXPERIMENTAL_PROVIDER`, to finch this time, to be able to create the cluster (see screenshot below for the whole process).

image

hope this helps and please let me know if I need to run specific commands for building it on WSL2/Windows.

File: site/content/docs/user/quick-start.md [hunk 4]
```diff
@@ -144,6 +144,9 @@ wait for 30 seconds, do `--wait 30s`, for 5 minutes do `--wait 5m`, etc.
 
 More usage can be discovered with `kind create cluster --help`.
 
+The kind can auto-detect the [docker], [podman], or [nerdctl] installed and choose the available one. If you want to turn off the auto-detect, use the environment variable `KIND_EXPERIMENTAL_PROVIDER=docker`, `KIND_EXPERIMENTAL_PROVIDER=podman` or `KIND_EXPERIMENTAL_PROVIDER=nerdctl` to
```
Comment (aojea) [inline site/content/docs/user/quick-start.md:147]: @nunix is your concern addressed by this comment?

**Comment by aojea** [comment]
waiting for resolving the last comment

File: site/content/docs/user/quick-start.md (hunk 4 above)
Comment (nunix) [reply site/content/docs/user/quick-start.md:147]: @aojea yes all good on my side for the comment.

I will try again the auto-detect when the 0.23 is out maybe, as what I was trying to test was actually the auto-detect. But it's maybe on my side and with the comment, it provides the needed help.

**Comment by aojea** [comment]
/lgtm
/approve
Thanks

**Pull request merged**